hs_core/models.py

```diff
@@ -2271,6 +2309,49 @@ def supports_delete_folder_on_zip(self, original_folder):
         """Check if resource allows the original folder to be deleted upon zip."""
         return True
 
+    @property
+    def storage_type(self):
+        if not self.is_federated:
+            return 'local'
+        userpath = '/' + os.path.join(
+            getattr(settings, 'HS_USER_IRODS_ZONE', 'hydroshareuserZone'),
+            'home',
+            getattr(settings, 'HS_LOCAL_PROXY_USER_IN_FED_ZONE', 'localHydroProxy'))
+        if self.resource_federation_path == userpath:
+            return 'user'
+        else:
+            return 'federated'
+
+    def is_folder(self, folder_path):
+        """Determine whether a given path (relative to resource root, including /data/contents/)
+           is a folder or not. Returns False if the path does not exist.
+        """
+        # logger = logging.getLogger(__name__)
+        # logger.debug("folder_path is {}".format(folder_path))
+        path_split = folder_path.split('/')
+        while path_split[-1] == '':
+            path_split.pop()
+        # logger.debug("path_split is {}".format(path_split))
+        # logger.debug("path_split[0:-1] is {}".format(path_split[0:-1]))
+        dir_path = u'/'.join(path_split[0:-1])
+        # logger.debug("dir_path is {}".format(dir_path))
+
+        # handles federation
+        irods_path = os.path.join(self.root_path, dir_path)
+        # logger.debug("irods_path is {}".format(irods_path))
+        istorage = self.get_irods_storage()
+        try:
+            listing = istorage.listdir(irods_path)
+            # logger.debug("listing is {}".format(listing))
+        except SessionException:
+            # logger.debug("exception during listing")
```

Comment:
It'd be good to log the irods specific exception here, since it should not happen in normal state, but when it happen, we should see the specific iRODS exception for debugging purpose.